modules/reindex/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecReindexIT.java

```diff
@@ -142,46 +160,7 @@ private void useCodec(String index, String codec) throws ExecutionException, Int
         assertAcked(client().admin().indices().prepareOpen(index).setWaitForActiveShards(1));
     }
 
-    private void flushAndRefreshIndex(String index) {
-
-        // Request is not blocked
-        for (String blockSetting : Arrays.asList(
-            SETTING_BLOCKS_READ,
-            SETTING_BLOCKS_WRITE,
-            SETTING_READ_ONLY,
-            SETTING_BLOCKS_METADATA,
-            SETTING_READ_ONLY_ALLOW_DELETE
-        )) {
-            try {
-                enableIndexBlock(index, blockSetting);
-                // flush
-                FlushResponse flushResponse = client().admin().indices().prepareFlush(index).setForce(true).execute().actionGet();
-                assertNoFailures(flushResponse);
-
-                // refresh
-                RefreshResponse refreshResponse = client().admin().indices().prepareRefresh(index).execute().actionGet();
-                assertNoFailures(refreshResponse);
-            } finally {
-                disableIndexBlock(index, blockSetting);
-            }
-        }
-    }
-
-    private void ingestDocs(String index, int nbDocs) throws InterruptedException {
```

Comment:
Can we use this instead? It helps in abstracting the code to ingest with each codec

Reply:
I'm not for-or-against using methods/factoring out to methods in tests. But, I do believe all the setup relating to a test should be self-contained and idempotent. As a general rule of thumb, I refrain from factoring out to methods before it's repeated too many times (at least three).

Reply:
In the case of ingestDocs function, we have wrapped both indexing and refresh calls to ensure segments are properly created. Essentially, we are wrapping or factoring out two operations in a single function call. 
The question is, would it be beneficial in the future if we keep the ingestion as is, and make it reusable (in case we want to add more tests)?